### PR TITLE
Changelog cleanup for 0.9 release

### PR DESCRIPTION
**Pull request by harrism**
Cleanup of changelog for 0.9 release. Moved a few things from improvement to new feature and vice versa. Fixed typos, inconsistencies, long lines.

### CODE REVIEW

File: CHANGELOG.md [hunk 1]
```diff
@@ -2,43 +2,40 @@
 
 ## New Features
 
+- PR #1993 Add CUDA-accelerated series aggregations: mean, var, std
 - PR #2111 IO Readers: Support memory buffer, file-like object, and URL inputs
 - PR #2012 Add `reindex()` to DataFrame and Series
 - PR #2097 Add GPU-accelerated AVRO reader
-- PR #2098 Align DataFrame and Series indices before executing binary ops
```
Comment (kkraus14) [inline CHANGELOG.md]: This should stay in new features because it introduced new functionality as opposed to being transparent

Comment (harrism) [reply CHANGELOG.md]: Can you say what the new feature is? The changelog entry is too vague for a new feature.

File: CHANGELOG.md [hunk 2]
```diff
@@ -56,29 +53,31 @@
 - PR #2131 Chunk rows logic added to csv_writer
 - PR #2129 Add functions in the Java API to support nullable column filtering
 - PR #2165 made changes to get_dummies api for it to be available in MethodCache
+- PR #2171 Add CodeCov integration, fix doc version, make --skip-tests work when invoking with source
 - PR #2184 handle remote orc files for dask-cudf
 - PR #2186 Add `getitem` and `getattr` style access to Rolling objects
 - PR #2168 Use cudf.Column for CategoricalColumn's categories instead of a tuple
-- PR #2193 Added more docuemtnation to `type_dispatcher` for specializing dispatched functors
+- PR #2193 DOC: cudf::type_dispatcher documentation for specializing dispatched functors
 - PR #2197 CSV Writer: Expose `chunksize` as a parameter for `to_csv`
```
Comment (kkraus14) [inline CHANGELOG.md]: Could be argued to be a new feature

File: CHANGELOG.md [hunk 3]
```diff
@@ -56,29 +53,31 @@
...
 - PR #2199 Better java support for appending strings
 - PR #2176 Added column dtype support for datetime, int8, int16 to csv_writer
 - PR #2209 Matching `get_dummies` & `select_dtypes` behavior to pandas
 - PR #2217 Updated Java bindings to use the new groupby API
 - PR #2214 DOC: Update doc instructions to build/install `cudf` and `dask-cudf`
-- PR #1993 Add iterator driven reduction for mean, var, std
 - PR #2220 Update Java bindings for reduction rename
 - PR #2224 implement isna, isnull, notna as dataframe functions
```
Comment (kkraus14) [inline CHANGELOG.md]: New features

File: CHANGELOG.md [hunk 4]
```diff
@@ -56,29 +53,31 @@
...
 - PR #2232 Move CodeCov upload from build script to Jenkins
 - PR #2236 Implement drop_duplicates for Series
```
Comment (kkraus14) [inline CHANGELOG.md]: New feature

File: CHANGELOG.md [hunk 5]
```diff
@@ -2,43 +2,41 @@
...
 - PR #2446 Add __array_function__ for index
 - PR #2437 ORC reader: Add 'use_np_dtypes' option
 - PR #2382 Add CategoricalAccessor add, remove, rename, and ordering methods
 - PR #2464 Native implement `__cuda_array_interface__` for Series/Index/Column objects
-- PR #2425 Allow rolling window to accept array-based user-defined functions
+- PR #2425 Rolling window now accepts array-based user-defined functions
 - PR #2442 Add __setitem__
-- PR #2449 Java column vector: added support for getting byte count of strings in a ColumnVector
+- PR #2449 Java support for getting byte count of strings in a ColumnVector
 - PR #2492 Add groupby.size() method
```
Comment (kkraus14) [inline CHANGELOG.md]: New feature

File: CHANGELOG.md (hunk 2 above)
Comment (kkraus14) [reply CHANGELOG.md]: It doesn't really affect functionality though, going to leave it as is.

**Pull request merged**